src/api_client/api_client.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import {
+	TESTNET_NETHASH,
+	TESTNET_NODES,
+	MAINNET_NETHASH,
+	MAINNET_NODES,
+} from 'constants';
+import * as resources from './resources';
+
+const defaultOptions = {
+	bannedNode: [],
+	version: '1.0.0',
+	minVersion: '>=1.0.0',
+	randomizeNode: true,
+};
+
+const commonHeaders = {
+	'Content-Type': 'application/json',
+	os: 'lisk-js-api',
+};
+
+const getHeaders = (nethash, version, minVersion) =>
+	Object.assign({}, commonHeaders, {
+		nethash,
+		version,
+		minVersion,
+	});
+
+export default class APIClient {
+	constructor(nodes, nethash, providedOptions = {}) {
+		if (!Array.isArray(nodes) || nodes.length <= 0) {
+			throw Error('Require nodes to be initialized.');
```

Comment:
Prefer `new` syntax. Also message could be improved. Maybe `APIClient requires nodes for initialization.`?